File: soccer/src/main/java/com/jk/soccer/model/local/converter/DateString.java
package com.jk.soccer.model.local.converter;

import androidx.room.TypeConverter;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class DateString {

    final static private SimpleDateFormat dateFormat
            = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US);

    @TypeConverter
    public static String dateToString(Date date){
        if (date == null)
            return "1999-12-31 23:59:59";
        return dateFormat.format(date);
    }

    @TypeConverter
    public static Date stringToDate(String string){
        try {
            return dateFormat.parse(string);
        } catch (Exception e){
            e.printStackTrace();
        }
        return null;
    }

}

File: price/src/main/java/com/jk/price/DateString.kt
package com.jk.price

import androidx.room.TypeConverter
import java.text.SimpleDateFormat
import java.util.*

class DateString {

    private val dateFormat = SimpleDateFormat("yyyy-MM-dd E요일", Locale.KOREA)

    @TypeConverter
    fun dateToString(date: Date?): String? {
        return if (date == null) dateFormat.format(Date()) else dateFormat.format(date)
    }

    @TypeConverter
    fun stringToDate(string: String?): Date? {
        try {
            return dateFormat.parse(string!!)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }

}